ChainSafe/gossamer
# fix(node): fix node restart

## Changes


- Implement pause functionality on state and sync services. The issue with our shutdown is that our services have different shutdown times and our state gets corrupted as some services are running while others stopped. To get around this I build the functionality to pause key services on shutdown to avoid the state being corrupted.

## Tests



1) Start a node 
`./bin/gossamer init --chain paseo` then 
`./bin/gossamer --chain paseo`

2) Stop node with sigint
3) Restart node and see if it resumes syncing `./bin/gossamer --chain paseo`

## Issues



#3740

## Primary Reviewer



@EclesioMeloJunior
